### PR TITLE
Add Security.txt

### PR DESCRIPTION
**Pull request by rcmaehl**
https://securitytxt.org/

### CODE REVIEW

**Comment by rcmaehl** [comment]
```
[warn] src/server/index.tsx
[warn] Code style issues found in the above file. Forgot to run Prettier?
```

Attempted to run a prettier using a few different online tools with no changes to the code. Not sure what it's complaining about.

File: src/server/handlers/security-handler.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import type { Response } from "express";
+
+export default async ({ res }: { res: Response }) => {
+  res.setHeader("content-type", "text/plain; charset=utf-8");
+
+  res.send(`Contact: mailto:security@lemmy.ml
```
Comment (tgxn) [inline src/server/handlers/security-handler.ts:6]: Ideally this would be driven by the URL you run Lemmy on, or a configuration option.

Comment (rcmaehl) [reply src/server/handlers/security-handler.ts:6]: True, however this is literally the first time I've ever done anything with typescript. Although, if you have code I can add in, I'll gladly do so.

Comment (SleeplessOne1917) [reply src/server/handlers/security-handler.ts:6]: `process.env.LEMMY_UI_HOST` should work for the instance URL.

File: src/server/index.tsx [hunk 2]
```diff
@@ -22,6 +23,7 @@ if (!process.env["LEMMY_UI_DISABLE_CSP"] && !process.env["LEMMY_UI_DEBUG"]) {
   server.use(setDefaultCsp);
 }
 
+server.get("/.well-known/security.txt", SecurityHandler);
```
Comment (dessalines) [inline src/server/index.tsx:26]: We'll need to make sure this gets handled by nginx too.

**Review by dessalines — COMMENTED** [review]
Husky didn't get set up properly, so its failing lint. You'll need to run prettier manually.

File: src/server/handlers/security-handler.ts (hunk 1 above)
Comment (rcmaehl) [reply src/server/handlers/security-handler.ts:6]: I can add it as an addition contact but I have concerns security@`process.env.LEMMY_UI_HOST` or admin@`process.env.LEMMY_UI_HOST` will reach the owner of the instance.

**Comment by rcmaehl** [comment]
Not sure why it requested a review from alectrocute but okay.

File: src/server/handlers/security-handler.ts (hunk 1 above)
Comment (rcmaehl) [reply src/server/handlers/security-handler.ts:6]: Added

**Comment by SleeplessOne1917** [comment]
@rcmaehl The actual changes seem good to me. @dessalines should merging this be held off until the nginx configs are updated to handle it?

**Comment by rcmaehl** [comment]
Ah so it requests a review every time I update the branch. New to big pull requests...

File: src/server/handlers/security-handler.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  Contact: mailto:admin@` + process.env.LEMMY_UI_HOST + `
```
Comment (dessalines) [inline src/server/handlers/security-handler.ts:7]: `LEMMY_UI_LEMMY_EXTERNAL_HOST` is the one with the actual domain.tld . The other ones are for the internal isomorphic server.

**Comment by dessalines** [comment]
@SleeplessOne1917 We can merge this before those nginx configs are done.

File: src/server/handlers/security-handler.ts (hunk 3 above)
Comment (rcmaehl) [reply src/server/handlers/security-handler.ts]: Done

**Pull request merged**